language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -10,498 +10,238 @@ import System.Environment.Blank
-
-    writeEsLintConfig :: FilePath -> IO ()
-    writeEsLintConfig dir = writeFileUTF8 (dir </> ".eslintrc.json") $ unlines
-      [ "{"
-      , "  \"parser\": \"@typescript-eslint/parser\","
-      , "  \"parserOptions\": {"
-      , "    \"project\": \"./tsconfig.json\""
-      , "  },"
-      , "  \"plugins\": ["
-      , "    \"@typescript-eslint\""
-      , "  ],"
-      , "  \"extends\": ["
-      , "    \"eslint:recommended\","
-      , "    \"plugin:@typescript-eslint/eslint-recommended\","
-      , "    \"plugin:@typescript-eslint/recommended\","
-      , "    \"plugin:@typescript-eslint/recommended-requiring-type-checking\""
-      , "  ],"
-      , "  \"rules\": {"
-      , "    \"@typescript-eslint/explicit-function-return-type\": \"off\","
-      , "    \"@typescript-eslint/no-inferrable-types\": \"off\""
-      , "  }"
-      , "}"
-      ]
-
-    writePackageJson :: FilePath -> IO ()
-    writePackageJson dir = let name = takeFileName dir in writeFileUTF8 (dir </> "package.json") $ unlines
-            ["{"
-            , "  \"private\": true,"
-            , "  \"name\": \"@daml2ts/" <> name <> "\","
-            , "  \"version\": \"" <> sdkVersion <> "\","
-            , "  \"description\": \"Produced by daml2ts\","
-            , "  \"license\": \"Apache-2.0\","
-            , "  \"dependencies\": {"
-            , "    \"@daml/types\": \"" <> sdkVersion <> "\","
-            , "    \"@mojotech/json-type-validation\": \"^3.1.0\""
-            , "  },"
-            , "  \"scripts\": {"
-            , "    \"build\": \"tsc --build\","
-            , "    \"lint\": \"eslint --ext .ts src/ --max-warnings 0\""
-            , "  },"
-            , "  \"devDependencies\": {"
-            , "    \"@typescript-eslint/eslint-plugin\": \"^2.11.0\","
-            , "    \"@typescript-eslint/parser\": \"^2.11.0\","
-            , "    \"eslint\": \"^6.7.2\","
-            , "    \"typescript\": \"~3.7.3\""
-            , "  }"
-            , "}"
-            ]
-
-    writeRootPackageJson :: FilePath -> String -> IO ()
-    writeRootPackageJson rootDir workspace =
-       writeFileUTF8 (rootDir </> "package.json") $ unlines
+    buildProject :: [String] -> IO ()
+    buildProject args = callProcessSilent damlc (["build"] ++ args)
+
+    daml2tsProject :: [FilePath] -> FilePath -> FilePath -> IO ()
+    daml2tsProject dars outDir packageJson = callProcessSilent daml2ts $ dars ++ ["-o", outDir, "-p", packageJson]
+
+    yarnProject :: [String] -> IO ()
+    yarnProject args = callProcessSilent yarn args
+
+    callProcessSilent :: FilePath -> [String] -> IO ()
+    callProcessSilent cmd args = do
+        (exitCode, out, err) <- readProcessWithExitCode cmd args ""
+        unless (exitCode == ExitSuccess) $ do
+          hPutStrLn stderr $ "Failure: Command \"" <> cmd <> " " <> unwords args <> "\" exited with " <> show exitCode
+          hPutStrLn stderr $ unlines ["stdout:", out]
+          hPutStrLn stderr $ unlines ["stderr: ", err]
```

Comment:
This doesn’t look right, you removed the `exitFailure`at the end.

Reply:
genuine bug. thank-you.